applications/CompressiblePotentialFlowApplication/custom_strategies/transonic_residualbased_incrementalupdate_static_scheme.h

```diff
@@ -0,0 +1,343 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Marco Antonio Zuñiga Perez
+//
+
+#if !defined(KRATOS_TRANSONIC_RESIDUALBASED_NEWTON_RAPHSON_STRATEGY )
+#define  KRATOS_TRANSONIC_RESIDUALBASED_NEWTON_RAPHSON_STRATEGY
+
+/* System includes */
+
+/* External includes */
+
+/* Project includes */
+#include "solving_strategies/schemes/scheme.h"
+#include "includes/define.h"
+#include "includes/variables.h"
+#include "utilities/entities_utilities.h"
+#include "custom_processes/compute_nodal_value_process.h"
+#include "compressible_potential_flow_application_variables.h"
+#include "fluid_dynamics_application_variables.h"
+#include "solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+///@}
+///@name Type Definitions
+///@{
+///@}
+///@name  Enum's
+///@{
+///@}
+///@name  Functions
+///@{
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class TransonicResidualBasedIncrementalUpdateStaticScheme
+ * @ingroup CompressiblePotentialFlowApplication
+ * @brief This class is a reimplementation of residualBasedIncrementalUpdateStaticScheme and is 
+ * used in a transonic potential simulation to update both the CRITICAL_MACH and 
+ * UPWIND_FACTOR_CONSTANT values, between nonlinear iterations, to a user-defined value when 
+ * a user-defined residual tolerance is reached. This update is done only once.
+ * @details The only operation done in this  scheme is the update of the database and values, no predict is done
+ * @tparam TSparseSpace The sparse space considered
+ * @tparam TDenseSpace The dense space considered
+ * @see Scheme
+ */
+template<class TSparseSpace,class TDenseSpace>
+class TransonicResidualBasedIncrementalUpdateStaticScheme : public ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace>
+{
+
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of TransonicResidualBasedIncrementalUpdateStaticScheme
+    KRATOS_CLASS_POINTER_DEFINITION(TransonicResidualBasedIncrementalUpdateStaticScheme);
+
+    typedef Scheme<TSparseSpace,TDenseSpace>                                 BaseSchemeType;
+    typedef ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace> BaseType;
+    typedef TransonicResidualBasedIncrementalUpdateStaticScheme<TSparseSpace, TDenseSpace> ClassType;
+
+    /// DoF array type definition
+    typedef typename BaseType::DofsArrayType                                  DofsArrayType;
```

Comment:
```suggestion
    using DofsArrayType = typename BaseType::DofsArrayType;
```
In this case it would be like this.